### PR TITLE
fix windows gnu build

### PR DESCRIPTION
**Pull request by thewh1teagle**
I tried to use this library on windows inside `msys2`, following `BUILDING.md` and got bunch of erros.
Fixed all of them and planning to improve the instructions as well. 
Checked in msys2 and it works with the examples of this repo.

https://github.com/tazz4843/whisper-rs/issues/87
https://github.com/tazz4843/whisper-rs/issues/65

Steps to build:
1. Download `msys2` from https://www.msys2.org/ and install
2. Install prerequisites from `ucrt64` terminal
```console
pacman --needed -S $MINGW_PACKAGE_PREFIX-{toolchain,make,cmake,rust,clang} git
```
3. Clone
```console
git clone -b fix/msys2-build --recursive https://github.com/thewh1teagle/whisper-rs.git
cd whisper-rs
```

4. Build
```console
cargo build
```

5. Try it
```console
wget -q --show-progress \
    https://huggingface.co/ggerganov/whisper.cpp/resolve/main/ggml-base.en.bin -O ggml-base.en.bin
cd examples/full_usage
cargo run 2830-3980-0043.wav ../../ggml-base.en.bin
```

### CODE REVIEW

File: src/whisper_grammar.rs [hunk 1]
```diff
@@ -5,8 +5,10 @@ use whisper_rs_sys::{
     whisper_gretype_WHISPER_GRETYPE_RULE_REF,
 };
 
+
 #[cfg_attr(not(windows), repr(u32))]
-#[cfg_attr(windows, repr(i32))] // windows being *special* again
+#[cfg_attr(target_env = "gnu", repr(u32))] // x86_64-pc-windows-gnu
```
[the remark this answers is not in the document]
Comment (thewh1teagle) [reply src/whisper_grammar.rs:10]: I'm sorry but I didn't understand how exactly.

Suggestion:
```rust
#[cfg_attr(any(not(windows), target_env = "gnu"), repr(u32))] // include windows-gnu
#[cfg_attr(all(windows, not(target_env = "gnu")), repr(i32))] // msvc being *special* again
```

[the remark this answers is not in the document]
Comment (tazz4843) [reply src/whisper_grammar.rs:10]: Yep that's what I would do

File: src/whisper_grammar.rs [hunk 2]
```diff
@@ -5,8 +5,10 @@ use whisper_rs_sys::{
     whisper_gretype_WHISPER_GRETYPE_RULE_REF,
 };
 
+
```
[the remark this answers is not in the document]
Comment (thewh1teagle) [reply src/whisper_grammar.rs]: I can't see it in my file. and rustfmt doesn't tell me to change that, weired. I'm on `windows`.

[the remark this answers is not in the document]
Comment (tazz4843) [reply src/whisper_grammar.rs]: Looks like it's gone now. Not sure what happened.

File: src/whisper_grammar.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (thewh1teagle) [reply src/whisper_grammar.rs]: Tested in `msys2` again just to make sure, works.

**Review by tazz4843 — APPROVED** [review]

**Comment by tazz4843** [comment]
Thanks :)

**Pull request merged**